src/controllers/secret.go

```diff
@@ -0,0 +1,194 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dana-team/platform-backend/src/types"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretController interface {
+	CreateSecret(types.CreateSecretRequest) (types.CreateSecretResponse, error)
+	GetSecrets(types.GetSecretsRequest) (types.GetSecretsResponse, error)
+	GetSecret(types.GetSecretRequest) (types.GetSecretResponse, error)
+	PatcnSecret(types.UpdateSecretRequest) (types.UpdateSecretResponse, error)
+	DeleteSecret(types.DeleteSecretRequest) (types.DeleteSecretResponse, error)
+}
+
+type secretController struct {
+	client *kubernetes.Clientset
+	ctx    context.Context
+	logger *zap.Logger
+}
+
+func NewSecretController(client *kubernetes.Clientset, context context.Context, logger *zap.Logger) SecretController {
+	return &secretController{
+		client: client,
+		ctx:    context,
+		logger: logger,
+	}
+}
```

Comment:
This is inconsistent with the naming on the [`namespace` controller](https://github.com/dana-team/platform-backend/blob/d4a737de4fff7edf0f4cf947f0229065836df185/src/controllers/namespace.go#L27). 

Open an issue to change the naming on that controller since we cannot reuse `New` since all controllers are in the same package right now.

Reply:
Yes, you're right. I'll take care of it in a new PR, we need to change it from `New` to `NewNamespaceController`.